File: keuzewijzer-hbo-deeltijd-ict-API/Models/Cohort.cs
using System.ComponentModel.DataAnnotations.Schema;

namespace keuzewijzer_hbo_deeltijd_ict_API.Models
{
    public class Cohort
    {
        public int Id { get; set; }
        public string Name { get; set; }
        public int Year { get; set; }
        public ICollection<SemesterItem>? SemesterItems { get; set; }
        public int? UserId { get; set; }

        [NotMapped]
        public User User { get; set; } = null!;

        public Cohort(int id, string name, int year, int userId)
        {
            Id = id;
            Name = name;
            Year = year;
            UserId = userId;
        }

        public Cohort(int id, string name, int year)
        {
            Id = id;
            Name = name;
            Year = year;

        }

        public Cohort()
        {

        }
    }
}

File: keuzewijzer-frontend/src/Models/Cohort.ts
import { Semester } from "./Semester";

export class Cohort {
    id: number;
    name: string;
    year: number;
    semesterItems: Semester[] | null;
    userId: number | null;
    // user: User; TODO: fix this

    constructor() {
        this.id = 0;
        this.name = '';
        this.year = 0;
        this.semesterItems = null;
        this.userId = null;
        // this.user = new User();
    }

}
